test/src/unit-cppapi-array.cc

```diff
@@ -724,51 +724,85 @@ TEST_CASE(
   }
 
   SECTION("MODIFY_EXCLUSIVE") {
```

Comment:
Could we make a fixture with the setup calls and split this into two separate tests. I would also like to flesh out the MODIFY_EXCLUSIVE test a little bit more. I would make a the commits consolidation optional (bool consolidate_commits = GENERATE(true, false);) and also have an optional vacuuming parameter (of course if vaccum is true and consolidate is false, skip the variation).

Another interesting test for completion could be to consolidate fragments written at timestamps 1 and 3 for example, then run fragment deletion from timestamps 2-4.